filesystem/src/web.ts

```diff
@@ -240,7 +247,13 @@ export class FilesystemWeb extends WebPlugin implements FilesystemPlugin {
     }
 
     if (occupiedEntry !== undefined) {
-      data = occupiedEntry.content + data;
+      if (occupiedEntry.content && !encoding) {
```

Comment:
Is it possible that occupiedEntry.content is undefined?

If it's not possible then we shouldn't check

If it's possible then there could be a case where there is no content and no encoding and it goes to the else and allows to write base64 without checking if it's base64